inc/cleantalk-settings.php

```diff
@@ -2679,3 +2684,36 @@ class="apbct-btn-as-link"
                 . __('change email', 'cleantalk-spam-protect') .
             '</button>)';
 }
+
+/**
+ * Implementation of service_update_local_settings functionality
+ */
+add_action('apbct_before_returning_settings', 'apbct__send_local_settings_to_api');
+
+function apbct__send_local_settings_to_api($settings)
+{
+    if (empty($settings) || !is_array($settings)) {
+        return;
+    }
+
+    global $apbct;
+
+    // Settings to JSON
+    $settings = json_encode($settings);
+
+    // Apikey
+    $apikey = $apbct->api_key;
+
+    // Hostname
+    $hostname = preg_replace('/^(https?:)?(\/\/)?(www\.)?/', '', get_site_url());
+
+    Helper::httpRequest(
+        'https://api-next.cleantalk.org',
+        array(
+            'method_name' => 'service_update_local_settings',
+            'auth_key' => $apikey,
+            'hostname' => $hostname,
+            'settings' => $settings
+        ),
+        'async');
+}
```

Comment:
You need to implement new method in Common\API class and call it here.

If you do so, the code will be 
```
function apbct__send_local_settings_to_api($settings){
API::method__service_update_local_settings( $PARAMS_BLA_BLA );
}
```
In this case, function apbct__send_local_settings_to_api() and hook doesn't make sense. The method sholud be called right from the end of apbct_settings__validate() function.

Reply:
I think that actions allow you to write more independent code
the method in the API did